hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/records/SubClusterPolicyConfiguration.java

```diff
@@ -127,36 +129,36 @@ public static SubClusterPolicyConfiguration newInstance(
 
   @Override
   public int hashCode() {
-    return 31 * getParams().hashCode() + getType().hashCode();
+    return new HashCodeBuilder().
+        append(this.getType()).
+        append(this.getQueue()).
+        append(this.getParams()).
+        toHashCode();
   }
 
   @Override
   public boolean equals(Object obj) {
+
     if (this == obj) {
       return true;
     }
-    if (obj == null) {
-      return false;
-    }
-    if (getClass() != obj.getClass()) {
-      return false;
-    }
-    SubClusterPolicyConfiguration other = (SubClusterPolicyConfiguration) obj;
-    if (!this.getType().equals(other.getType())) {
-      return false;
-    }
-    if (!this.getParams().equals(other.getParams())) {
+
+    if (obj == null || getClass() != obj.getClass()) {
       return false;
     }
-    return true;
+
+    SubClusterPolicyConfiguration right = (SubClusterPolicyConfiguration) obj;
+
+    return new EqualsBuilder()
+        .append(this.getType(), right.getType())
+        .append(this.getQueue(), right.getQueue())
+        .append(this.getParams(), right.getParams())
+        .isEquals();
   }
 
   @Override
   public String toString() {
-    StringBuilder sb = new StringBuilder();
-    sb.append(getType())
-        .append(" : ")
-        .append(getParams());
-    return sb.toString();
+    return "SubClusterPolicyConfiguration [getType() = " + getType() + ","
+        + "getQueue() = " + getQueue() + ",getParams() = " +  getParams() + "]";
   }
 }
```

Comment:
No new line at end of file.

Reply:
Thank you very much for your help reviewing the code, I will fix it.